modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java

```diff
@@ -1,141 +1,346 @@
+	 * @param type the class of the target {@link JsonMessage}
+	 * @param <T>  the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the buffer contents does not contain the necessary information to successfully
+	 *                         construct the specified {@link JsonMessage}
 	 */
-	@JsonProperty(access = JsonProperty.Access.READ_ONLY)
-	default String className() {
-		return getClass().getName();
+	static <T extends JsonMessage> T from(Buffer json, Class<T> type) throws DecodeException {
+		return Json.CODEC.fromBuffer(json, type);
 	}
 
 	/**
-	 * Returns the Json-representation of the message.
+	 * Constructs a {@link JsonMessage} from a JSON {@link String}.
 	 *
-	 * @return json representation of message
+	 * @param json the JSON {@link String} that contains the necessary information to construct
+	 *             the specified {@link JsonMessage}
+	 * @param type the class of the target {@link JsonMessage}
+	 * @param <T>  the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the JSON string does not contain the necessary information to successfully
+	 *                         construct the specified {@link JsonMessage}
 	 */
-	default JsonObject json() {
+	static <T extends JsonMessage> T from(String json, Class<T> type) throws DecodeException {
+		return Json.CODEC.fromString(json, type);
+	}
+
+	/**
+	 * Constructs a {@link JsonMessage} from a plain {@link Object}.
+	 *
+	 * @param json the plain {@link Object} that contains the necessary information to construct
+	 *             the specified {@link JsonMessage}
+	 * @param type the class of the target {@link JsonMessage}
+	 * @param <T>  the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the plain object does not contain the necessary information to successfully
+	 *                         construct the specified {@link JsonMessage}
+	 */
+	static <T extends JsonMessage> T from(Object json, Class<T> type) throws DecodeException {
+		return Json.CODEC.fromValue(json, type);
+	}
+
+	/**
+	 * Constructs a {@link JsonMessage} from a Vert.x EventBus {@link Message} body.
+	 *
+	 * @param message the message which body contains the necessary information to construct
+	 *                the specified {@link JsonMessage}
+	 * @param type    the class of the target {@link JsonMessage}
+	 * @param <T>     the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the raw message body does not contain the necessary information to successfully
+	 *                         construct the specified {@link JsonMessage}
+	 */
+	static <T extends JsonMessage> T from(Message<?> message, Class<T> type) throws DecodeException {
+		return from(message.body(), type);
+	}
+
+	///
+	/// SYNCHRONOUS ENCODING SECTION
+	///
+
+	/**
+	 * @deprecated Use {@link #toJsonObject()} instead.
```

Comment:
Why do you deprecate this? Isn't it actually more convenient to use than `toJsonObject()`?
I just want to understand the benefit of switching to the new version...

Reply:
Ah, I get it. You want to be able to differentiate between the multiple new versions...
Maybe add this to the javadocs of `json()` and don't just deprecate it 😉

Reply:
Done in [`c077263` (#532)](https://github.com/wuespace/telestion-core/pull/532/commits/c077263e80848027c4139ceb9b40a260d548f92b)